client/ayon_deadline/lib.py

```diff
@@ -43,3 +47,338 @@ def get_instance_job_envs(instance) -> "dict[str, str]":
         env = dict(sorted(env.items()))
 
     return env
+
+
+class DeadlineKeyValueVar(dict):
+    """
+
+    Serializes dictionary key values as "{key}={value}" like Deadline uses
+    for EnvironmentKeyValue.
+
+    As an example:
+        EnvironmentKeyValue0="A_KEY=VALUE_A"
+        EnvironmentKeyValue1="OTHER_KEY=VALUE_B"
+
+    The keys are serialized in alphabetical order (sorted).
+
+    Example:
+        >>> var = DeadlineKeyValueVar("EnvironmentKeyValue")
+        >>> var["my_var"] = "hello"
+        >>> var["my_other_var"] = "hello2"
+        >>> var.serialize()
+
+
+    """
+    def __init__(self, key):
+        super(DeadlineKeyValueVar, self).__init__()
+        self.__key = key
+
+    def serialize(self):
+        key = self.__key
+
+        # Allow custom location for index in serialized string
+        if "{}" not in key:
+            key = key + "{}"
+
+        return {
+            key.format(index): "{}={}".format(var_key, var_value)
+            for index, (var_key, var_value) in enumerate(sorted(self.items()))
+        }
+
+
+class DeadlineIndexedVar(dict):
+    """
+
+    Allows to set and query values by integer indices:
+        Query: var[1] or var.get(1)
+        Set: var[1] = "my_value"
+        Append: var += "value"
+
+    Note: Iterating the instance is not guarantueed to be the order of the
+          indices. To do so iterate with `sorted()`
+
+    """
+    def __init__(self, key):
+        super(DeadlineIndexedVar, self).__init__()
+        self.__key = key
```

Comment:
Why is used dunder score? (Can be potentially dangerous as it does not have the same way).

Reply:
Nevermind, it is just moved from other file, out of this PR.